Apps/GatewayApi/src/Validations/UserProfileValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.GatewayApi.Validations
+{
+    using FluentValidation;
+    using HealthGateway.Common.Data.Models;
+
+    /// <summary>
+    /// Validates <see cref="UserProfile"/> instances.
+    /// </summary>
+    public class UserProfileValidator : AbstractValidator<UserProfile>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserProfileValidator"/> class.
+        /// </summary>
+        public UserProfileValidator()
+        {
+            this.RuleFor(profile => profile.SmsNumber)
+                .Must(smsNumber => string.IsNullOrEmpty(smsNumber) || PhoneNumberValidator.IsValid(smsNumber));
```

Comment:
Can we move the age validator to here?

Reply:
Currently the age validation is attached to the `IUserProfileService`. This is used in the UserProfileController's Validate endpoint and the `CreateUserProfile` method.

Should I honour the Open/Closed principle of the interface and leave the service method as is by rather adding some "duplication" to the validator, by taking the steps from the [ValidateMinimumAge](https://github.com/bcgov/healthgateway/blob/dev/Apps/GatewayApi/src/Services/UserProfileService.cs#L413) and strip out the `RequestResult` elements, thus its a duplication of process rather than a total duplicate.

Reply:
Deemed inappropriate to address